packages/handlersjs-core/lib/logging/logger.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { Logger } from './logger';
+import { LoggerLevel } from './logger-level';
+
+describe('Logger', () => {
+
+  class MockLogger extends Logger {
+
+    log(level: LoggerLevel, typeName: string, message: string, data?: unknown): void {
+
+      throw new Error('Method not implemented.');
+
+    }
+
+    constructor(minimumLevel: LoggerLevel, minimumLevelPrintData: LoggerLevel) {
+
+      super(minimumLevel, minimumLevelPrintData);
+
+    }
+
+  }
+
+  const logger = new MockLogger(LoggerLevel.info, LoggerLevel.info);
+
```

Comment:
```suggestion
const paramCheck = (fn: string, typeName: string, message: string) => {

    if (!typeName) {

      switch (fn) {

        case 'info':
          return expect(() => logger.info(typeName, message)).toThrow('Typename should be set');
        case 'debug':
          return expect(() => logger.debug(typeName, message)).toThrow('Typename should be set');
        case 'warn':
          return expect(() => logger.warn(typeName, message)).toThrow('Typename should be set');
        case 'error':
          return expect(() => logger.error(typeName, message)).toThrow('Typename should be set');

      }

    }

    if (!message) {

      switch (fn) {

        case 'info':
          return expect(() => logger.info(typeName, message)).toThrow('Message should be set');
        case 'debug':
          return expect(() => logger.debug(typeName, message)).toThrow('Message should be set');
        case 'warn':
          return expect(() => logger.warn(typeName, message)).toThrow('Message should be set');
        case 'error':
          return expect(() => logger.error(typeName, message)).toThrow('Message should be set');

      }

    }

  };

```

Reply:
This doesn't really look better to me, see comment below.

Reply:
See my comments on that 😉